EXPERIMENTS.md

```diff
@@ -0,0 +1,188 @@
+values, we can increase our confidence in the specification and its properties.
+While this restriction to small parameter values is dictated by the practicality
+of our approach, it is informally supported by the "small scope hypothesis,"
+which was popularized by tools such as [Alloy][].
+
+## 3. All figures in one place
+
+The following table summarizes the experimental figures in one place:
+
+| Experiment | Specification | Property   | Time    | Memory |
+|-----------:|---------------|------------|---------|--------|
+| 4.1        | `Spec 2`      | Row 1      | TBD     | TBD    |
+| 5.1        | `Spec 3`      | Row 2      | TBD     | TBD    |
+| 5.2        | `Spec 3`      | Row 3      | TBD     | TBD    |
+| 6.1        | `Spec 4`      | Row 4      | TBD     | TBD    |
+| 6.2        | `Spec 4`      | Row 5      | TBD     | TBD    |
+
+## 4. Model checking Spec 2
+
+<!-- TODO: Thomas, please add a description of the experiments with MC_ffg_examples.tla -->
+
+### 4.1. Bounded model checking of Accountable Safety
+
+In this experiment, we are checking whether Accountable Safety holds true for
+`Spec 2`. Since `Spec 2` presents a snapshot of a single validator and does not
+make any steps, we have to check this property only in the initial states.
+
+```sh
+$ cd ./spec
+$ JVM_ARGS=-Xmx20G apalache-mc check --length=0 \
+  --inv=AccountableSafety MC_ffg.tla
+```
+
+Note that we have to extend the default JVM heap size from 4G to 20G.
+
+Since `Spec 2` has high complexity, this experiment takes long. **TODO:** how
+long.
+
+## 5. Model checking Spec 3
+
+### 5.1. Bounded model checking of Accountable Safety
+
+In this experiment, we are checking whether Accountable Safety holds true for up
+to 10 steps of `Spec 3`. Since `Spec 3` specifies the abstract state machine of
+a single validator (alternatively, the network view), we have to check
+executions up to some bound.
+
+```sh
+$ cd ./abstract-spec
+$ JVM_ARGS=-Xmx20G apalache-mc check --inv=AccountableSafety MC_ffg.tla
+```
+
+This experiment takes about **TODO**.
+
+### 5.2. Randomized symbolic execution for accountable safety
+
+To obtain results faster, albeit at the cost of completeness, we analyze 100
+random symbolic runs, each consisting of up to 10 steps:
+
+```sh
+$ cd ./abstract-spec
+$ JVM_ARGS=-Xmx20G apalache-mc simulate \
+  simulate --max-run=100 --length=10 --timeout-smt=10800 \
+  --inv=AccountableSafety MC_ffg.tla
+```
+
+In addition, we bound the running time of individual SMT queries to 3 hours. If
+the timeout is reached, the query is skipped. This command typically finds
+counterexamples much quicker than `check`. Therefore, when `simulate` does not
+find counterexamples, it gives us relatively good confidence that the property
+holds true.
+
+This experiment takes about **TODO** on a single node. The interesting thing is
+that we can parallelize the enumeration of 100 runs across multiple cores. For
+instance, we try 5 symbolic runs on 20 CPU cores.
```

Comment:
There is some qualifier missing in the last sentence. Maybe @Kukovec has a better idea.

```suggestion
that we can parallelize the enumeration of 100 runs across multiple cores. For
instance, we use 20 CPU cores to check 5 symbolic runs each.
```